Issue 200/migrate calendars and campaigns page to mui

nehahirve opened the pull request:
transition calendars to MUI and add (some) responsiveness to search drawer and add responsiveness to navigation

I can't change the screenshots for some reason but the breadcrumbs have also been migrated

Fixes #202 and #203

richardolsson commented:
This is going to be so good! I'm glad we're switching to MUI. 😊 

I'm going to review the code soon, but first I wanted to point out some visual issues that arose when I tried this with the live data, see screenshot below:

* Some campaign bars on the left render as `0`
* Some events seem to render _behind_  the calendar
* The inner scrollbar of the calendar overlaps the month/week view selector (is the select list part of the scrolling container?)
* There are three scrollbars: Two full-height and then the calendar. There should only be one IMO
* The events seem to be positioned very far down in each grid slot. Are they absolutely positioned based on start time? I don't think they should be in the month view, just a simple stacked list

![image](https://user-images.githubusercontent.com/550212/122516188-be8d8f80-d00e-11eb-8169-2fc7153d6902.png)

richardolsson commented on src/components/MonthCalendar.tsx:
```diff
@@ -84,44 +87,30 @@ const MonthCalendar = ({ campaigns, events, onFocusDate, focusDate }: MonthCalen
                                                     alignItems: 'center',
-                                                    background: campaign?.color || '#d3d3d3',
-                                                    color: getContrastColor(campaign?.color|| '#d3d3d3'),
+                                                    background: campaign?.color || grey[400],
+                                                    color: getContrastColor(campaign?.color|| grey[400]),
```
The `grey[400]` thing is nice and reminds me of Flutter!

richardolsson commented on src/components/OrganizeCampaignTabs.tsx:
```diff
@@ -1,39 +1,81 @@
+                />
             </Tabs>
-        </View>
+            <TabPanel index="summary" value={ value }>
+                { children }
+            </TabPanel>
+            <TabPanel index="calendar" value={ value }>
+                { children }
+            </TabPanel>
+            <TabPanel index="insights" value={ value }>
+                { children }
+            </TabPanel>
```
Why three different instances of the same children? I feel like it should be enough to just use a single `TabPanel` for layout (which wouldn't need to be a separate component) without the `index` and `value` props, with the children inside it.

richardolsson commented on src/components/SearchDrawer.tsx:
```diff
@@ -46,13 +64,13 @@ const SearchDrawer = (): JSX.Element | null => {
                     />
-                    <View isHidden={ !drawerOpen }>
-                        <Text>
+                    <Box display={ drawerOpen ? 'block' : 'none' }>
```
Is there any MUI component we could use instead of the custom solution for the `SearchDrawer`?

richardolsson commented on src/components/WeekCalendar.tsx:
```diff
@@ -63,101 +79,61 @@ const WeekCalendar = ({ campaigns, events, focusDate, onFocusDate }: WeekCalenda
-                    zIndex: 1,
-
-                }}>
+            <Box { ...{ ref: calendarWrapper } } data-testid="calendar-wrapper" height={ 1 } overflow="scroll" width={ 1 }>
```
Correct me if I'm wrong (because it's quite difficult to read) but isn't there just a single top-level `<Box>` here? If that's the case, there should be no need to wrap in a fragment.

richardolsson commented on src/pages/_app.tsx:
```diff
@@ -21,6 +22,9 @@ declare global {
     interface Window {
         __reactRendered: boolean;
     }
+    interface Date {
+        getWeekNumber: () => number;
+    }
```
Maybe this should actually be moved to the `polyfills.ts` module where the function is defined? I think that might make it easier for a reader to understand what's going on. 😊

richardolsson reviewed (changes requested):
Nice! 👍 

Some issues that should be fixed below, but mostly I have questions about things I don't understand or would like to hear your thoughts on. 😊

nehahirve commented on src/components/MonthCalendar.tsx:
```diff
@@ -84,44 +87,30 @@ const MonthCalendar = ({ campaigns, events, onFocusDate, focusDate }: MonthCalen
-                                    width: '100%',
-                                }}>
+                            <Box key={ index } bgcolor={ isInRange(currentDate, firstMonthDay, lastMonthDay) ? grey[300] : grey[200] } data-testid={ `griditem-${index}` } m={ 0.5 } position="relative">
+                                <Typography>
+                                    <Box p={ 1 }>
```
The docs said that the children prop is of 'node' type and it seems to apply to the content inside box even if not directly text. Don't know if this is standard use ==> I will place it inside Box here because it only applies to the text in this case, anyway, not multiple nodes.

nehahirve commented:
i couldn't figure out any way to not make the srollbar overlap that top bar -- tried out a lot of things, for now i moved the select a little inward so it looks better, but unless there's some kind of css property i am missing i'm at a loss. i've added it to the refactor list for now
